tests/apidocs/util.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+import os
+from django.test.client import RequestFactory
+from openapi_core import create_spec
+from openapi_core.contrib.django import DjangoOpenAPIRequest, DjangoOpenAPIResponse
+from openapi_core.validation.response.validators import ResponseValidator
+
+from sentry.utils import json
+from sentry.testutils import APITestCase
+
+
+class APIDocsTestCase(APITestCase):
+    def create_schema(self):
+        # get the generated json file
+        path = os.path.join(os.path.dirname(__file__)) + "/openapi-derefed.json"
+        with open(path, "r") as json_file:
+            data = json.load(json_file)
+            # remove DSN and update server to testserver
+            del data["components"]
+
+            return create_spec(data)
+
+    def create_test_response(self, path):
+        request = RequestFactory().get(path)
+        request.headers = {}
+        request._current_scheme_host = "https://sentry.io/"
+        return request
+
+    def validate_schema(self, url):
+        self.login_as(user=self.user)
+        validator = ResponseValidator(self.create_schema())
+
+        response = self.client.get(url)
```

Comment:
I don't know that this is going to be best way to do things. It's not always going to be a `GET` and there are other arguments that made need to be passed in. 

I think I would prefer to keep the creation of the response in the individual test files. I think it will make it easier to see what's actually being tested (what method) and that way we can keep the `self.login_as` in there too. 

To me it makes sense for this to take the `response` as an argument. If you need the url for the request, then you could have both.